<!-- FroboMind -->

<launch>
		<node pkg="rosbag" name="sensors" type="play" args="--start=65 --rate=1 --clock $(find sdu_hudem_2014)/test.bag">
			<remap from="/rosbag/clock" to="/clock"/>
		</node>

	<group ns="fmProcessors">
		<!--node pkg="sdu_hudem_2014" name="wptnav" type="show_navigation_status_node.py"  output="screen">
			<param name="wptnav_status_sub" value="/fmInformation/wptnav_status" />
		</node-->

		<node pkg="odometry_gnss_pose_estimator" name="pose_estimator" type="pose_estimator_node.py" output="screen" >
			<remap from="/rosbag/clock" to="/clock"/>
			<param name="odom_sub" value="/fmKnowledge/odometry" />
			<param name="imu_sub" value="/fmInformation/imu" />
			<param name="gga_sub" value="/fmInformation/gpgga_tranmerc" />
			<param name="pose_pub" value="/fmKnowledge/pose_test" />
			<param name="pose_status_pub" value="/fmKnowledge/pose_status" />
			<param name="odometry_timeout" value="0.5" /> 
			<param name="gnss_timeout" value="2.0" /> 
			<param name="imu_timeout" value="0.5" />
			<param name="absolute_orientation_timeout" value="30" />
			<param name="odometry_distance_variance" value="0.000001" />
			<param name="odometry_angular_variance" value="0.000001" /> 
			<param name="gnss_angular_variance" value="0.0001" />
			<param name="solution_required" value="4" />
		</node>

		<node pkg="pose_2d_estimator" name="pose_2d_estimator" type="pose_2d_estimator_node.py" output="screen" respawn="true">
			<param name="odom_sub" value="/fmKnowledge/odometry" />
			<param name="imu_sub" value="/fmInformation/imu" />
			<param name="gga_sub" value="/fmInformation/gpgga_tranmerc" />
			<param name="pose_pub" value="/fmKnowledge/pose_ref" />
			<param name="publish_relative_pose" type="boolean" value="false" />
			<param name="odometry_distance_variance" value="0.000001" /> <!-- [m/update] -->
			<param name="odometry_angular_variance" value="0.000001" /> <!-- [rad/update] -->
		</node>

		<node pkg="frobit_plot" name="armadillo3_plot" type="frobit_plot_node.py" output="screen" >
			<param name="pose_sub" value="/fmKnowledge/pose_ref" />
			<param name="gnss_sub" value="/fmInformation/gpgga_tranmerc" />
			<param name="odom_sub" value="/fmKnowledge/encoder_odom" />
			<param name="wptnav_status_sub" value="/fmInformation/wptnav_status" />
			<param name="plot_pose_track" value="true" />
			<param name="plot_gnss_track" value="false" />
			<param name="plot_odometry_track" value="false" />
			<param name="plot_pose_yaw" value="true" />
			<param name="map_trackpoint_threshold" value="0.02" />
			<param name="map_max_trackpoints" value="1000" />
			<param name="map_minimum_size" value="6.0" />
			<param name="map_easting_offset" value="0.0" />
			<param name="map_northing_offset" value="0.0" />
			<param name="map_update_frequency" value="10.0" />
			<param name="map_title" value="Reference" />
			<param name="map_window_size" value="5.0" />
			<param name="avatar_extension_front" value="0.4" />
			<param name="avatar_extension_rear" value="0.4" />
			<param name="avatar_extension_lateral" value="0.50" />
			<param name="avatar_reverse" value="true" />
		</node>

		<node pkg="frobit_plot" name="armadillo3_plot_test" type="frobit_plot_node.py" output="screen" >
			<param name="pose_sub" value="/fmKnowledge/pose_test" />
			<param name="gnss_sub" value="/fmInformation/gpgga_tranmerc" />
			<param name="odom_sub" value="/fmKnowledge/encoder_odom" />
			<param name="wptnav_status_sub" value="/fmInformation/wptnav_status" />
			<param name="plot_pose_track" value="true" />
			<param name="plot_gnss_track" value="false" />
			<param name="plot_odometry_track" value="false" />
			<param name="plot_pose_yaw" value="true" />
			<param name="map_trackpoint_threshold" value="0.02" />
			<param name="map_max_trackpoints" value="1000" />
			<param name="map_minimum_size" value="6.0" />
			<param name="map_easting_offset" value="0.0" />
			<param name="map_northing_offset" value="0.0" />
			<param name="map_update_frequency" value="10.0" />
			<param name="map_title" value="TEST" />
			<param name="map_window_size" value="5.0" />
			<param name="avatar_extension_front" value="0.4" />
			<param name="avatar_extension_rear" value="0.4" />
			<param name="avatar_extension_lateral" value="0.50" />
			<param name="avatar_reverse" value="true" />
		</node>
	</group>
</launch>

